tests/borsh_encoding.rs

```diff
@@ -106,96 +106,83 @@ impl BorshToken {
 
 /// Encode a signed integer
 fn encode_int(width: u16, value: &BigInt, buffer: &mut Vec<u8>) {
-    match width {
-        8 => {
-            let val = value.to_i8().unwrap();
-            buffer.extend_from_slice(&val.to_le_bytes());
-        }
-        16 => {
-            let val = value.to_i16().unwrap();
-            buffer.extend_from_slice(&val.to_le_bytes());
-        }
-        32 => {
-            let val = value.to_i32().unwrap();
-            buffer.extend_from_slice(&val.to_le_bytes());
-        }
-        64 => {
-            let val = value.to_i64().unwrap();
-            buffer.extend_from_slice(&val.to_le_bytes());
-        }
-        128 => {
-            let val = value.to_i128().unwrap();
-            buffer.extend_from_slice(&val.to_le_bytes());
-        }
-        n if n <= 256 => {
-            let mut val = value.to_signed_bytes_le();
-            let byte_width = (width / 8) as usize;
-            match val.len().cmp(&byte_width) {
-                Ordering::Greater => {
-                    while val.len() > byte_width {
-                        val.pop();
-                    }
+    if width <= 8 {
```

Comment:
Just out of curiosity, why did you change this to be a `else if` chain?

Reply:
In the `match` statement, I wanted to match the width to an exact number, and now I have to deal with a range. If you know how to do this using `match`, please, let me know and I will change my code.

Reply:
[You can try something like this](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=986405b4429057f9e2af1a275ae283cc)

Reply:
That looks nice! I'm going to change my code to match your suggestion.